// ngx-text-highlight/ngx-text-highlight.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', '@angular/common'], factory) :
	(factory((global['ngx-text-highlight'] = {}),global.core,global.common));
}(this, (function (exports,core,common) { 'use strict';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var HighlightDirective = (function () {
    function HighlightDirective(el) {
        this.el = el;
        this.separatedText = [];
        this.separatedSearchedText = [];
        this.final = '';
        this.splitFlag = '';
        this.matchFlag = '';
        this.spanStart = '<span class="ngx-text-highlight">';
        this.spanEnd = '</span>';
        this.caseSensitive = false;
    }
    Object.defineProperty(HighlightDirective.prototype, "content", {
        get: /**
         * @return {?}
         */
        function () { return this._content; },
        set: /**
         * @param {?} content
         * @return {?}
         */
        function (content) {
            this._content = content;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HighlightDirective.prototype, "searchTerm", {
        get: /**
         * @return {?}
         */
        function () { return this._searchTerm; },
        set: /**
         * @param {?} searchTerm
         * @return {?}
         */
        function (searchTerm) {
            this._searchTerm = searchTerm;
            this.highlight();
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    HighlightDirective.prototype.highlight = /**
     * @return {?}
     */
    function () {
        this.final = '';
        if (!this.caseSensitive) {
            this.splitFlag = 'i';
            this.matchFlag = 'gi';
        }
        else {
            this.splitFlag = '';
            this.matchFlag = 'g';
        }
        this.searchPattern = new RegExp(this._searchTerm, this.splitFlag);
        this.matchpattern = new RegExp(this._searchTerm, this.matchFlag);
        if (this._searchTerm !== undefined && this._searchTerm != null && this._searchTerm.length > 0 && this._searchTerm[0] !== '.') {
            this.separatedText = this._content.split(this.searchPattern);
            this.separatedSearchedText = this._content.match(this.matchpattern);
            if (this.separatedSearchedText != null && this.separatedSearchedText.length > 0) {
                for (var /** @type {?} */ i = 0; i < this.separatedText.length; i++) {
                    if (i <= this.separatedSearchedText.length - 1) {
                        this.final += this.separatedText[i] + this.spanStart + this.separatedSearchedText[i] + this.spanEnd;
                    }
                    else {
                        this.final += this.separatedText[i];
                    }
                }
            }
            if (this.final.length > 0) {
                this.el.nativeElement.innerHTML = this.final;
            }
            else {
                this.el.nativeElement.innerText = this._content;
            }
        }
        else {
            this.el.nativeElement.innerText = this._content;
        }
    };
    HighlightDirective.decorators = [
        { type: core.Directive, args: [{
                    selector: '[ngxTextHighlight]'
                },] },
    ];
    /** @nocollapse */
    HighlightDirective.ctorParameters = function () { return [
        { type: core.ElementRef, },
    ]; };
    HighlightDirective.propDecorators = {
        "content": [{ type: core.Input, args: ['content',] },],
        "searchTerm": [{ type: core.Input, args: ['searchTerm',] },],
        "caseSensitive": [{ type: core.Input },],
    };
    return HighlightDirective;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NgHighlightModule = (function () {
    function NgHighlightModule() {
    }
    /**
     * @return {?}
     */
    NgHighlightModule.forRoot = /**
     * @return {?}
     */
    function () {
        return {
            ngModule: NgHighlightModule
        };
    };
    NgHighlightModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        HighlightDirective
                    ],
                    exports: [
                        HighlightDirective
                    ]
                },] },
    ];
    /** @nocollapse */
    NgHighlightModule.ctorParameters = function () { return []; };
    return NgHighlightModule;
}());

exports.NgHighlightModule = NgHighlightModule;
exports.HighlightDirective = HighlightDirective;

Object.defineProperty(exports, '__esModule', { value: true });

})));
